cloudwatchlogs/node.js/cloudwatchlogs_lambda.js

```diff
@@ -2,79 +2,112 @@
 // Remember to change the hostname and path to match your collection API and specific HTTP-source endpoint
 // See more at: https://help.sumologic.com/APIs/01Collector_Management_API
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////
-var sumoEndpoint = 'https://collectors.sumologic.com/receiver/v1/http/<XXX>'
+var sumoEndpoint = 'https://collectors.sumologic.com/receiver/v1/http/<XXX>';
+ 
+// Format used to parse out log formats
+// Example: 2016-11-10T23:11:54.523Z	108af3bb-a79b-11e6-8bd7-91c363cc05d9    some message
+var consoleFormatRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z\t(\w+?-\w+?-\w+?-\w+?-\w+)\t(.*)/;
 
 var https = require('https');
 var zlib = require('zlib');
 var url = require('url');
 
-exports.handler = function(event, context) {
-	    var urlObject = url.parse(sumoEndpoint);
-	    
-	    var options = { 'hostname': urlObject.hostname,
-				'path': urlObject.pathname,
-				'method': 'POST'
-			};
+exports.handler = function (event, context) {
+    var urlObject = url.parse(sumoEndpoint);
+    
+    var options = {
+        'hostname': urlObject.hostname,
+        'path': urlObject.pathname,
+        'method': 'POST'
+    };
+    
+    var zippedInput = new Buffer(event.awslogs.data, 'base64');
+    
+    zlib.gunzip(zippedInput, function (e, buffer) {
+        if (e) {
+            context.fail(e);
+        }
+        
+        var awslogsData = JSON.parse(buffer.toString('ascii'));
```

Comment:
@kumoroku : it's base64 first. After that it seems to be ascii:  http://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html#LambdaFunctionExample